src/hooks/useSearchSubmit.ts

```diff
@@ -22,8 +23,16 @@ export function useSearchSubmit(): (query: string) => void {
       // Orders, transactions, tokens, batches
       const path = pathAccordingTo(query)
       const pathPrefix = prefixNetwork ? `${prefixNetwork}/${path}` : `${path}`
-
-      query && query.length > 0 && history.push(`/${pathPrefix}/${query}`)
+      if (pathPrefix === 'address') {
+        if (web3) {
+          web3.eth.ens
+            .getAddress(query)
+            .then((res) => res && res.length > 0 && history.push(`/${pathPrefix}/${res}`))
+            .catch(() => history.push(`/address/null`))
```

Comment:
For the ENS case, when it cannot be resolved, it should just take you to the default. This is kind of unrelated of this PR (so out of it's scope), however, I mention it because `address/null` is not a good redirection imo

A good one could be:
![image](https://user-images.githubusercontent.com/2352112/137523509-5114f7d7-7a71-4dd3-b77a-6a51e4b0b5fd.png)

Reply:
In my latest commit, I route to `/address/${query}` so let me know if it should be something yet different.  If it's a valid ENS domain, it will just show no records (since there will be valid ETH address behind it) or else the generic "Page not found" where an invalid address is provided as the query parameter.